File: 3174-minimum-number-of-changes-to-make-binary-string-beautiful/3174-minimum-number-of-changes-to-make-binary-string-beautiful.java
class Solution {
    public int minChanges(String s) {
        int n = s.length();
        int count = 0;
        char ch = s.charAt(0);
        int consecutive = 0;
        for (int i = 0; i < n; i++) {
            if (s.charAt(i) == ch) {
                consecutive++;
                continue;
            }
            if (consecutive % 2 == 0) {
                consecutive = 1;
            } else {
                consecutive = 0;
                count++;
            }
            ch = s.charAt(i);
        }
        return count;
    }
}

File: 3174-minimum-number-of-changes-to-make-binary-string-beautiful/3174-minimum-number-of-changes-to-make-binary-string-beautiful.cpp
class Solution {
public:
    int minChanges(string s) {
        int n = s.length();
        if(n == 0) return 0;
        int ans = 0;
        int count = 0;
        char prev = s[0];
        for(int i = 0; i < n; i++) {
            if(prev == s[i]) count++;
            else {
                if(count % 2 == 0) {
                    count = 1;
                    prev = s[i];
                }
                else  {
                    count = 0;
                    ans += 1;
                    prev = s[i];
                }
            }
        }
        if(count % 2) ans += 1;
        return ans;
    }
};